src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Objects;
+
+/**
+ * A settings for the multi thread mode.
+ * @author Andrew Kuchev
+ */
+public class ThreadModeSettings implements Serializable {
+    /** A checker module name. */
+    public static final String CHECKER_MODULE_NAME = Checker.class.getSimpleName();
+
+    /** A multi thread checker module name. */
+    public static final String MULTI_THREAD_CHECKER_MODULE_NAME =
+            Checker.class.getSimpleName();
+
+    /** A three walker module name. */
+    public static final String TREE_WALKER_MODULE_NAME = TreeWalker.class.getSimpleName();
+
+    /** A multi thread three walker module name. */
+    public static final String MULTI_THREAD_TREE_WALKER_MODULE_NAME =
+            TreeWalker.class.getSimpleName();
+
+    private static final long serialVersionUID = 1L;
+
+    /** The checker threads number. */
+    private final int checkerThreadsNumber;
+    /** The tree walker threads number. */
+    private final int treeWalkerThreadsNumber;
+
+    /**
+     * Initializes the multi thread mode configuration.
+     * @param checkerThreadsNumber the Checker threads number
+     * @param treeWalkerThreadsNumber the TreeWalker threads number
+     */
+    public ThreadModeSettings(int checkerThreadsNumber, int treeWalkerThreadsNumber) {
+        this.checkerThreadsNumber = checkerThreadsNumber;
+        this.treeWalkerThreadsNumber = treeWalkerThreadsNumber;
+    }
+
+    /**
+     * Gets the number of threads for the Checker module.
+     * @return the number of threads for the Checker module.
+     */
+    public int getCheckerThreadsNumber() {
+        return checkerThreadsNumber;
+    }
+
+    /**
+     * Gets the number of threads for the TreeWalker module.
+     * @return the number of threads for the TreeWalker module.
+     */
+    public int getTreeWalkerThreadsNumber() {
+        return treeWalkerThreadsNumber;
+    }
+
+    /**
+     * Creates a settings object which will enable single thread mode for
+     * Checker and TreeWalker modules.
+     * @return created settings
+     */
+    public static ThreadModeSettings getSingleThreadMode() {
+        return new ThreadModeSettings(1, 1);
+    }
+
+    /**
+     * Resolves the module name according to the thread settings.
+     * @param name The original module name.
+     * @return resolved module name.
+     */
+    public final String resolveName(String name) {
+        if (CHECKER_MODULE_NAME.equals(name)
+                && getCheckerThreadsNumber() > 1) {
+            throw new IllegalArgumentException(
+                    "Multi thread mode for Checker module is not implemented");
+        }
+        if (TREE_WALKER_MODULE_NAME.equals(name)
+                && getTreeWalkerThreadsNumber() > 1) {
+            throw new IllegalArgumentException(
+                    "Multi thread mode for TreeWalker module is not implemented");
+        }
+
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Taken from https://github.com/checkstyle/checkstyle/pull/4420#discussion_r123917283 .

> > When are we going to have 2 instances to compare?

> A test is using PowerMock to mock a constructor (https://github.com/checkstyle/checkstyle/pull/4420/files#diff-06fde9f319ac6e31f8a808d413fe1f8eR593) and it is using equals to compare arguments

If you assign `ThreadModeSettings.getSingleThreadMode()` to a variable for all uses in that test, would that then alleviate the need for an `equals` method?
If not, I will need to debug the specific case to see what is the reason for this.